### PR TITLE
clusters can define allowing failures to not halt processing

### PR DESCRIPTION
**Pull request by c1freitas**
# What 

Clusters can now define allowable failures during processing. By default, any failure during the generation of files would halt all processing. Cluster can now set a `metadata.allowFailure: true` to override this process. If set to true, a warning event will be emitted and processing will continue for other resources/clusters.

# Testing
Added new failure test cluster in fixtures. Setting `metadata.allowFailure: false` for this cluster fails multiple integration tests.

### CODE REVIEW

File: test/unit/lib/generator.spec.js [hunk 1]
```diff
@@ -81,6 +83,7 @@ describe("Generator", () => {
 				expect(localConfig.svc).to.exist;
 				expect(localConfig).to.not.equal(clusterDef.configuration());
 				expect(localConfig.name).to.equal("auth");
+				console.log("LocalCOnfig %j", localConfig);
```
Comment (chesleybrown) [inline test/unit/lib/generator.spec.js]: Typo `LocalCOnfig %j`

Do you mean to leave these console.logs in your tests though?

File: test/functional/lib/deploymentizer.spec.js [hunk 2]
```diff
@@ -25,6 +25,11 @@ describe("Deploymentizer", () => {
 						save: true,
 						conf: conf
 					});
+				// multiple events will get fired for failure cluster.
+				deployer.events.on(deployer.events.WARN, function(message) {
+					console.log("WARN::::" + message);
```
Comment (chesleybrown) [inline test/functional/lib/deploymentizer.spec.js]: Maybe use `expect` to verify the messages you are getting are what you want? Just using console.log requires manual review?

Comment (c1freitas) [inline test/functional/lib/deploymentizer.spec.js]: With events there is no guarantee that you will get all events before the test completes.

File: test/unit/lib/generator.spec.js (hunk 1 above)
Comment (c1freitas) [inline test/unit/lib/generator.spec.js]: Doh!

**Pull request merged**